tests/ValueFormatters/TimeFormatterTest.php

```diff
@@ -112,8 +81,15 @@ public function validProvider() {
 		$options = new FormatterOptions();
 
 		foreach ( $tests as $expected => $args ) {
-			$timeValue = new TimeValue( $args[0], $args[1], $args[2], $args[3], $args[4], $args[5] );
-			$argLists[] = array( $timeValue, $expected, $options );
+			$timestamp = $args[0];
+			$precision = isset( $args[1] ) ? $args[1] : TimeValue::PRECISION_DAY;
+			$calendarModel = isset( $args[2] ) ? $args[2] : $gregorian;
+
+			$argLists[] = array(
+				new TimeValue( $timestamp, 0, 0, 0, $precision, $calendarModel ),
+				$expected,
+				$options
+			);
```

Comment:
#49 contains a major refactoring of this. I did not wanted to repeat everything, this patch here contains only very basic stuff.